src/components/ui/dialog/custom-dialog.tsx

```diff
@@ -24,30 +29,40 @@ export const CustomDialog = ({
   description,
   showCloseButton = true,
   showSubmitButton = true,
+  showLinkButton = false,
   closeButtonLabel,
   submitButtonLabel,
+  linkButtonLabel,
   variant,
   size,
   maxWidthMin,
   onClose,
   onSubmit,
   loading,
+  linkTo,
+  dialogNoButton,
+  dialogYesButton,
 }: CustomDialogProps) => {
   return (
     <Dialog open={open} variant={variant} size={size} maxWidthMin={maxWidthMin}>
       <Dialog.Title>{title}</Dialog.Title>
       <Dialog.Description>{description}</Dialog.Description>
       <Dialog.Actions>
         {showCloseButton && (
-          <Button variant='primaryOutline' onClick={onClose}>
+          <Button variant='primaryOutline' onClick={onClose} testId={dialogNoButton}>
             {closeButtonLabel ?? "No"}
           </Button>
         )}
         {showSubmitButton && (
-          <Button variant='primary' onClick={onSubmit} loading={loading}>
+          <Button variant='primary' onClick={onSubmit} loading={loading} testId={dialogYesButton}>
             {submitButtonLabel ?? "Yes"}
           </Button>
         )}
+        {showLinkButton && (
+          <LinkButton to={linkTo ?? ""} testId={dialogYesButton}>
+            {linkButtonLabel ?? "Yes"}
+          </LinkButton>
+        )}
```

Comment:
For this one maybe we could use `Button` instead of `LinkButton` for those using this dialog component. Just create a function to navigate and pass that function to the onSubmit button.